scripts/sysman/virtualFunctionManagement.yml

```diff
@@ -0,0 +1,252 @@
+      name: "timestamp"
+      desc: "[out] Wall clock time when the activeCounterValue was sampled."
+--- #--------------------------------------------------------------------------
+type: function
+desc: "Get handle of virtual function modules"
+class: $sDevice
+name: EnumActiveVirtualFunctionExp
+details:
+    - "The application may call this function from simultaneous threads."
+    - "The implementation of this function should be lock-free."
+params:
+    - type: $s_device_handle_t
+      name: hDevice
+      desc: "[in] Sysman handle of the device."
+    - type: "uint32_t*"
+      name: pCount
+      desc: |
+            [in,out] pointer to the number of components of this type.
+            if count is zero, then the driver shall update the value with the total number of components of this type that are available.
+            if count is greater than the number of components of this type that are available, then the driver shall update the value with the correct number of components.
+    - type: "$s_vf_handle_t*"
+      name: pVFhandle
+      desc: |
+            [in,out][optional][range(0, *pCount)] array of handle of components of this type.
+            if count is less than the number of components of this type that are available, then the driver shall only retrieve that number of component handles.
+--- #--------------------------------------------------------------------------
+type: function
+desc: "Get virtual function management properties"
+class: $sVFManagement
+name: GetVFPropertiesExp
+details:
+    - "The application may call this function from simultaneous threads."
+    - "The implementation of this function should be lock-free."
+params:
+    - type: $s_vf_handle_t
+      name: hVFhandle
+      desc: "[in] Handle for the VF component."
+    - type: $s_vf_properties_exp_t*
+      name: pProperties
+      desc: "[in,out] Will contain VF properties."
+--- #--------------------------------------------------------------------------
+type: function
+desc: "Get memory activity stats for each available memory types associated with Virtual Function (VF)"
+class: $sVFManagement
+name: GetVFMemoryUtilizationExp
+details:
+    - "The application may call this function from simultaneous threads."
+    - "The implementation of this function should be lock-free."
+params:
+    - type: $s_vf_handle_t
+      name: hVFhandle
+      desc: "[in] Handle for the component."
+    - type: "uint32_t*"
+      name: pCount
+      desc: |
+            [in,out] Pointer to the number of VF memory stats descriptors.
+              - if count is zero, the driver shall update the value with the total number of memory stats available.
+              - if count is greater than the total number of memory stats available, the driver shall update the value with the correct number of memory stats available.
+              - The count returned is the sum of number of VF instances currently available and the PF instance.
+    - type: $s_vf_util_mem_exp_t*
```

Comment:
Mem type can be MEM_TYPE_SYSTEM and DEVICE right ?
So each VF can have MEM_TYPE_SYSTEM and DEVICE  utilizations seperately ?
Please share whether we would be able to retrieve both information for each VF and the PF using this API.

Reply:
@aravindksg  @sanilkumar0  this clarification still seems valid.
Could you see if there is a misunderstanding.
Thanks

Reply:
yes, it is intended that from API PoV we have capability to expose system and device mem utilizations for each VF from the PF.

Reply:
Since the count is"count returned is the sum of number of VF instances currently available and the PF instance", please share how this API would return both in a single call ?

Reply:
@sanilkumar0 I think Santosh's point is valid, in these APIs, the input is a VF handle so we return the utilization stats wrt that VF. So for PF the expectation might be use existing `zesMemoryGetState` call for gathering utilization info. @sanilkumar0 could you please verify that.

We might need to clarify the description around these APIs if that's the case.